entrypoint.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+if [ -f /tmp/phockup.lockfile ]; then
```

Comment:
Isn't this simply defeating the purpose of the lock file if you actually delete it before execution?

Reply:
no - it gets deleted once upon start of the container, not multiple times during its execution. 

Once the container is started, the cron job takes care of the continued execution of phockup. 
When starting up the container, there can't be any running instance of phockup - that's why we can delete the lockfile. As you could stop a docker container at any time, you could possibly do that during a phockup execution - and that would leave a lockfile behind.

Reply:
I think this does not resolve my concern. What will happen if the cron job is set to be executed every minute and I dump multiple GBs of photos which cannot be processed for a minute? I guess a second cron job will be started and we will have two processes working on the same files.

Reply:
Perhaps I am still not getting your point so let me describe mine in other words:

A Docker container is like a pretty simple VM. It boots up, executes a command and then keeps running as long as the execution of that command does not terminate.

Before my change, that command was a single call to phockup. Phockup ran, it exited, the container shut down.

In order to do something repeatedly or continuously inside a docker container without it shutting down, you need a command that keeps running indefinitely. Otherwise, the docker container would just stop. No repetition, nothing - that would be it ...

That is the reason why I created entrypoint.sh. It sets up a few prerequisites and then fires up a command that keeps running and does not exit.

That command is `crond -f -d 8`. Before executing the command we ensure the following: 
- The definition for our cron job has been set up using the parameters that the user provided. 
- Perhaps our Docker container was shutdown ungracefully and we still have a lock on phockup executions. It is safe to remove this lock now, before we "go into the loop" and let crond execute phockup in user defined intervals. 

Now lets take a look at what cron executes. The cron job that was set up executes phocker but we inserted the tool flock on top of that. Flock ensures that cron jobs do not overlap and only one phockup instance is active at the same time. Whether or not we removed any preexisting lock before entering this indefinite state of cron-based phockup executions does not matter now. If you are interested, I found an [article]((https://ma.ttias.be/prevent-cronjobs-from-overlapping-in-linux/)) explaining the use of flock together with cron - it also hints at alternatives that could have been used instead.